coreneuron/network/partrans.cpp

```diff
@@ -129,45 +129,69 @@ void nrnthread_v_transfer(NrnThread* _nt) {
+        if (!nt->compute_gpu) {
+            continue;
+        }
+
         TransferThreadData& ttd = transfer_thread_data_[tid];
 
-        size_t n_src_indices = ttd.src_indices.size();
-        size_t n_src_gather = ttd.src_gather.size();
-        NrnThread* nt = nrn_threads + tid;
-        if (n_src_indices) {
+        if (!ttd.src_indices.empty()) {
+            size_t n_src_indices = ttd.src_indices.size();
             int* src_indices = ttd.src_indices.data();
+            cnrn_target_copyin(src_indices, n_src_indices);
+
+            size_t n_src_gather = ttd.src_gather.size();
             double* src_gather = ttd.src_gather.data();
-            nrn_pragma_acc(enter data copyin(src_indices[0:n_src_indices]) if(nt->compute_gpu))
-            nrn_pragma_acc(enter data create(src_gather[0:n_src_gather]) if(nt->compute_gpu))
-            // clang-format off
-            nrn_pragma_omp(target enter data map(to: src_indices [0:n_src_indices])
-                                             map(alloc: src_gather[0:n_src_gather])
-                                             if(nt->compute_gpu))
-            // clang-format on
+            nrn_pragma_acc(enter data create(src_gather[0:n_src_gather]))
+            nrn_pragma_omp(target enter data map(alloc: src_gather[0:n_src_gather]))
         }
 
         if (ttd.insrc_indices.size()) {
             int* insrc_indices = ttd.insrc_indices.data();
             size_t n_insrc_indices = ttd.insrc_indices.size();
-            nrn_pragma_acc(
-                enter data copyin(insrc_indices [0:n_insrc_indices]) if (nt->compute_gpu))
-            // clang-format off
-            nrn_pragma_omp(target enter data map(to: insrc_indices[0:n_insrc_indices])
-                                             if(nt->compute_gpu))
-            // clang-format on
+            cnrn_target_copyin(insrc_indices, n_insrc_indices);
+        }
+    }
+}
+
+void nrn_partrans::delete_gap_indices_from_device() {
+    if (insrcdspl_) {
+        int n_insrc_buf = insrcdspl_[nrnmpi_numprocs];
+        cnrn_target_delete(insrc_buf_, n_insrc_buf);
+    }
+    for (int tid = 0; tid < nrn_nthread; ++tid) {
+        const NrnThread* nt = nrn_threads + tid;
+        if (!nt->compute_gpu) {
+            continue;
+        }
+
+        TransferThreadData& ttd = transfer_thread_data_[tid];
+
+        if (!ttd.src_indices.empty()) {
+            size_t n_src_indices = ttd.src_indices.size();
+            int* src_indices = ttd.src_indices.data();
+            cnrn_target_delete(src_indices, n_src_indices);
+
+            size_t n_src_gather = ttd.src_gather.size();
+            double* src_gather = ttd.src_gather.data();
+            cnrn_target_delete(src_gather, n_src_gather);
+        }
+
+        if (!ttd.insrc_indices.empty()) {
+            int* insrc_indices = ttd.insrc_indices.data();
+            size_t n_insrc_indices = ttd.insrc_indices.size();
+            cnrn_target_delete(insrc_indices, n_insrc_indices);
```

Comment:
I would have written
```suggestion
            cnrn_target_delete(ttd.insrc_indices.data(), ttd.insrc_indices.size());
```
and similar elsewhere, but I don't have a strong opinion.